packages/server/lib/controllers/auth/postUnauthenticated.ts

```diff
@@ -136,6 +137,30 @@ export const postPublicUnauthenticated = asyncWrapper<PostPublicUnauthenticatedA
             return;
         }
 
+        const customValidationResponse = await validateConnection({
+            connection: updatedConnection.connection,
+            config,
+            environment,
+            account,
+            logContextGetter
+        });
+
+        if (customValidationResponse.isErr()) {
+            void logCtx.error('Connection failed custom validation', { error: customValidationResponse.error });
+            await logCtx.failed();
+
+            // since this is an invalid connection, delete it with no trace of it
+            await connectionService.hardDelete(updatedConnection.connection.id);
```

Comment:
I can't be hard deleted like this, because those endpoints are upserting the connection might already exists and thus have enabled scripts/lots of records/jobs etc.

Reply:
Not sure I understand. The hook `connectionCreated` that instantiates the sync etc happens [later](https://github.com/NangoHQ/nango/blob/khaliq/nan-2858-spec-custom-verification-endpoints-as-a-on-events-script/packages/server/lib/controllers/auth/postUnauthenticated.ts#L172). But maybe you mean something else?